src/main/java/org/broadinstitute/barclay/help/scanners/FieldScanner.java

```diff
@@ -0,0 +1,52 @@
+package org.broadinstitute.barclay.help.scanners;
+
+import jdk.javadoc.doclet.DocletEnvironment;
+import org.broadinstitute.barclay.utils.Utils;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.util.ElementScanner14;
+import java.lang.reflect.Field;
+
+/**
+ * An {@link ElementScanner14} for finding the {@link Element} that corresponds to a given {@link Field}, given
+ * an Element that represents the class in which the field is declared.
+ *
+ * It's possible in some cases for a field to not have a corresponding element within a given enclosing
+ * element, even if the field is declared within the class represented by the (enclosing) element, i.e.,
+ * this can happen if the field is initialized by instantiating an anonymous class.
+ */
+public class FieldScanner extends ElementScanner14<Void, Void> {
```

Comment:
Same question about making this a scanner 17.

Reply:
See the other comments.